spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java

```diff
@@ -78,15 +94,59 @@ public GcpSpannerAutoConfiguration(GcpSpannerProperties gcpSpannerProperties,
 				: projectIdProvider.getProjectId();
 		this.instanceId = gcpSpannerProperties.getInstanceId();
 		this.databaseName = gcpSpannerProperties.getDatabase();
+		this.numRpcChannels = gcpSpannerProperties.getNumRpcChannels();
+		this.prefetchChunks = gcpSpannerProperties.getPrefetchChunks();
+		this.minSessions = gcpSpannerProperties.getMinSessions();
+		this.maxSessions = gcpSpannerProperties.getMaxSessions();
+		this.maxIdleSessions = gcpSpannerProperties.getMaxIdleSessions();
+		this.writeSessionsFraction = gcpSpannerProperties.getWriteSessionsFraction();
+		this.keepAliveIntervalMinutes = gcpSpannerProperties
+				.getKeepAliveIntervalMinutes();
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	public SpannerOptions spannerOptions() {
-		return SpannerOptions.newBuilder()
+	public SpannerOptions spannerOptions(SessionPoolOptions sessionPoolOptions) {
+		Builder builder = SpannerOptions.newBuilder()
 				.setProjectId(this.projectId)
 				.setHeaderProvider(new UsageTrackingHeaderProvider(this.getClass()))
-				.setCredentials(this.credentials).build();
+				.setCredentials(this.credentials);
+		if (this.numRpcChannels > 0) {
```

Comment:
I think you meant `>=0` as the default is -1?

Reply:
(this applies for the rest of the properties too)

Reply:
fixed!